pkg/tfvars/aws/aws.go

```diff
@@ -85,7 +88,15 @@ func TFVars(vpc string, privateSubnets []string, publicSubnets []string, masterC
 		Type:                    *rootVolume.EBS.VolumeType,
 		VPC:                     vpc,
 		PrivateSubnets:          privateSubnets,
-		PublicSubnets:           publicSubnets,
+		PublishStrategy:         string(publish),
+	}
+
+	if len(publicSubnets) == 0 {
+		if cfg.VPC != "" {
+			cfg.PublicSubnets = &[]string{}
```

Comment:
I feel like we should support installer-provided VPCs with internal publishing, where the user expects to create a VPN into the cluster before the waiting-for-Kube-API bit times out.  In that case, we'd want to switch on `publish` here.  But that would definitely be a corner case.  Thoughts?

Reply:
> I feel like we should support installer-provided VPCs with internal publishing,

Well that would be possible in this setup, as in that case public subnets will be set as nil ie create for me and vpc is also empty....

> Thoughts?

Currently the validation is not in place, and there will be following up PRs for that, and create vpc + internal will be explicitly not allowed as we specified in the design. The reason is the UX for ipi in this case is going to be too over the place that we don't want to support it right now.